app/controllers/concerns/wp_preview_tools.rb

```diff
@@ -0,0 +1,19 @@
+#
+# This concern may be mixed into a controller to add common functinality
+# for preview rendering, and preview token validation.
+#
+module WpPreviewTools
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def self.validate_preview_token(wp_post_model)
+      return wp_post_model.status == 'publish'
+      head :forbidden unless params[:token] == token(wp_post_model)  # TODO (cies): check if we have params[] here!
```

Comment:
Unreachable code detected.<br>Line is too long. [115/80]<br>Unnecessary spacing detected.